src/main/java/at/hannibal2/skyhanni/features/event/hoppity/HoppityRabbitTheFishChecker.kt

```diff
@@ -0,0 +1,102 @@
+package at.hannibal2.skyhanni.features.event.hoppity
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryUpdatedEvent
+import at.hannibal2.skyhanni.features.inventory.chocolatefactory.ChocolateFactoryAPI
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.anyMatches
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.SoundUtils
+import net.minecraft.client.Minecraft
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.lwjgl.input.Keyboard
+
+@SkyHanniModule
+object HoppityRabbitTheFishChecker {
+
+    //<editor-fold desc="Patterns">
+    /**
+     * REGEX-TEST: Chocolate Breakfast Egg
+     * REGEX-TEST: Chocolate Lunch Egg
+     * REGEX-TEST: Chocolate Dinner Egg
+     */
+    private val mealEggInventoryPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "inventory.mealegg.name",
+        "(?:§.)*Chocolate (?:Breakfast|Lunch|Dinner) Egg.*",
+    )
+
+    /**
+     * REGEX-TEST: §cRabbit the Fish
+     */
+    private val rabbitTheFishItemPattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "item.rabbitthefish",
+        "(?:§.)*Rabbit the Fish",
+    )
+
+    /**
+     * REGEX-TEST: Click to open Chocolate Factory!
+     */
+    private val openCfSlotLorePattern by ChocolateFactoryAPI.patternGroup.pattern(
+        "inventory.mealegg.continue",
+        "(?:§.)*Click to open Chocolate Factory!"
+    )
+    //</editor-fold>
+
+    private val config get() = SkyHanniMod.feature.event.hoppityEggs
+    private var rabbitTheFishIndex: Int? = null
+
+    @SubscribeEvent
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+
+        rabbitTheFishIndex?.let {
+            InventoryUtils.getItemsInOpenChest()[it] highlight LorenzColor.RED
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryUpdate(event: InventoryUpdatedEvent) {
```

Comment:
is there a reason we use inventory update over inventory fully opdated event here?